saturation.go

```diff
@@ -0,0 +1,148 @@
+package raft
+
+import (
+	"math"
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+// saturationMetric measures the saturation (percentage of time spent working vs
+// waiting for work) of an event processing loop, such as runFSM. It reports the
+// saturation as a gauge metric (at most) once every reportInterval.
+//
+// Callers must instrument their loop with calls to sleeping and working, starting
+// with a call to sleeping.
+//
+// Note: it is expected that the caller is single-threaded and is not safe for
+// concurrent use by multiple goroutines.
```

Comment:
This read a bit weird (caller is not safe for concurrent use?) so I attempted to simplify:
```suggestion
// Note: saturationMetric is not thread-safe and should not be used concurrently
// by multiple goroutines
```